crates/webidl/src/util.rs

```diff
@@ -34,35 +38,380 @@ pub fn mdn_doc(class: &str, method: Option<&str>) -> String {
+    fn to_syn_type(&self, _record: &FirstPassRecord<'src>, pos: TypePosition)
+        -> Option<syn::Type>
+    {
+        Some(array("u8", pos))
+    }
+}
+
+impl<'src> ToSynType<'src> for weedle::term::Uint8ClampedArray {
+    fn to_syn_type(&self, _record: &FirstPassRecord<'src>, pos: TypePosition)
+        -> Option<syn::Type>
+    {
+        Some(array("u8", pos))
+    }
+}
+
+impl<'src> ToSynType<'src> for weedle::term::Uint16Array {
+    fn to_syn_type(&self, _record: &FirstPassRecord<'src>, pos: TypePosition)
+        -> Option<syn::Type>
+    {
+        Some(array("u16", pos))
+    }
+}
+
+impl<'src> ToSynType<'src> for weedle::term::Uint32Array {
+    fn to_syn_type(&self, _record: &FirstPassRecord<'src>, pos: TypePosition)
+        -> Option<syn::Type>
+    {
+        Some(array("u32", pos))
+    }
+}
+
+impl<'src> ToSynType<'src> for weedle::types::Type<'src> {
+    fn to_syn_type(&self, record: &FirstPassRecord<'src>, pos: TypePosition)
+        -> Option<syn::Type>
+    {
+        use weedle::types::NonAnyType::*;
+        let single = match self {
+            Type::Single(s) => s,
+            Type::Union(_) => return None,
+        };
+
+        let ty = match single {
+            // `any` becomes `::wasm_bindgen::JsValue`.
+            SingleType::Any(_) => {
+                let path = vec![rust_ident("wasm_bindgen"), rust_ident("JsValue")];
+                return Some(simple_path_ty(path))
+            }
+            SingleType::NonAny(other) => other,
+        };
+
+        match ty {
+            Boolean(s) => s.to_syn_type(record, pos),
+            Octet(s) => s.to_syn_type(record, pos),
+            Byte(s) => s.to_syn_type(record, pos),
+            Identifier(s) => s.to_syn_type(record, pos),
+            Integer(s) => s.to_syn_type(record, pos),
+            FloatingPoint(s) => s.to_syn_type(record, pos),
+
+            Float32Array(s) => s.to_syn_type(record, pos),
+            Float64Array(s) => s.to_syn_type(record, pos),
+            Int8Array(s) => s.to_syn_type(record, pos),
+            Int16Array(s) => s.to_syn_type(record, pos),
+            Int32Array(s) => s.to_syn_type(record, pos),
+            Uint8Array(s) => s.to_syn_type(record, pos),
+            Uint8ClampedArray(s) => s.to_syn_type(record, pos),
+            Uint16Array(s) => s.to_syn_type(record, pos),
+            Uint32Array(s) => s.to_syn_type(record, pos),
+
+            DOMString(s) => s.to_syn_type(record, pos),
+            ByteString(s) => s.to_syn_type(record, pos),
+            USVString(s) => s.to_syn_type(record, pos),
+
+            // Support for these types is not yet implemented, so skip
+            // generating any bindings for this function.
+            ArrayBuffer(_)
+            | DataView(_)
+            | Error(_)
+            | FrozenArrayType(_)
+            | Object(_)
+            | Promise(_)
+            | RecordType(..)
+            | Sequence(_)
+            | Symbol(_) => {
+                None
```

Comment:
When we will add an actual support for these types, it will be quiet likely that adding support for stringifying will be forgotten. So, as it should be very trivial, adding support for turning these types into strings now might be a better approach?